components/ILIAS/Authentication/classes/class.ilAuthLogoutPageEditorGUI.php

```diff
@@ -0,0 +1,398 @@
+<?php
+
+/**
+ * This file is part of ILIAS, a powerful learning management system
+ * published by ILIAS open source e-Learning e.V.
+ *
+ * ILIAS is licensed with the GPL-3.0,
+ * see https://www.gnu.org/licenses/gpl-3.0.en.html
+ * You should have received a copy of said license along with the
+ * source code, too.
+ *
+ * If this is not the case or you just want to try ILIAS, you'll find
+ * us at:
+ * https://www.ilias.de
+ * https://github.com/ILIAS-eLearning
+ *
+ *********************************************************************/
+
+declare(strict_types=1);
+
+/**
+ * @ilCtrl_isCalledBy ilAuthLogoutPageEditorGUI: ilObjAuthSettingsGUI
+ * @ilCtrl_Calls      ilAuthLogoutPageEditorGUI: ilLogoutPageGUI
+ */
+class ilAuthLogoutPageEditorGUI
+{
+    private ilCtrlInterface $ctrl;
+    private ilLanguage $lng;
+    private ilGlobalTemplateInterface $tpl;
+    private ilTabsGUI $tabs;
+    private ilToolbarGUI $toolbar;
+    private ilRbacSystem $rbacsystem;
+    private ilSetting $setting;
+    private ilErrorHandling $ilErr;
+    private ?ilPropertyFormGUI $form;
+    private \ILIAS\HTTP\Services $http;
+    private \ILIAS\Refinery\Factory $refinery;
+    private \ILIAS\UI\Factory $ui_factory;
+    private \ILIAS\UI\Renderer $ui_renderer;
+
+    private int $ref_id;
+    private ilAuthLogoutPageEditorSettings $settings;
+    private ?ilSetting $logoutSettings = null;
```

Comment:
According to our code style, new variables/properties should be written in `snake_case`.